x-pack/legacy/plugins/apm/server/routes/create_api/index.ts

```diff
@@ -65,7 +65,7 @@ export function createApi() {
           body: bodyRt && 'props' in bodyRt ? t.exact(bodyRt) : fallbackBodyRt
         };
 
-        router[routerMethod](
+        (router[routerMethod] as RouterHandler<typeof routerMethod>)(
```

Comment:
the necessity to declare Method in generic type creates additional work for the NP plugins. what benefits did they get in return?